include/fiction/algorithms/physical_design/post_layout_optimization.hpp

```diff
@@ -441,12 +469,17 @@ layout_coordinate_path<Lyt> get_path_and_obstruct(Lyt& lyt, const tile<Lyt>& sta
  * @tparam Lyt Cartesian obstruction gate-level layout type.
  * @param lyt 2DDWave-clocked cartesian obstruction gate-level layout.
  * @param old_pos Old position of the gate to be moved.
+ * @param start The start time in milliseconds.
+ * @param timeout The timeout limit in milliseconds.
+ * @param timeout_limit_reached Reference to a boolean flag that is set to `true` if the timeout is reached.
  * @param planar_optimization Only allow relocation if a crossing-free wiring can be found. Defaults to false.
  * @return `true` if the gate was moved successfully, `false` otherwise.
  */
 template <typename Lyt>
 bool improve_gate_location(Lyt& lyt, const tile<Lyt>& old_pos, const tile<Lyt>& max_non_po,
-                           const uint64_t max_gate_relocations, const bool planar_optimization = false) noexcept
+                           const uint64_t                                       max_gate_relocations,
+                           const std::chrono::high_resolution_clock::time_point start, const uint64_t timeout,
+                           bool& timeout_limit_reached, const bool planar_optimization = false) noexcept
```

Comment:
I would prevent passing timeout parameters around and instead make them private members of the `post_layout_optimization_impl` class.

Particularly here, you have to pass three different variables for timeout tracking. That's excessive and makes it easy to use this function wrong.